blockchains/erc20/lib/fetch_events.js

```diff
@@ -13,27 +13,32 @@ const QNT_contract = '0x4a220e6096b25eadb88358cb44068a3248254675';
 const WETH_contract = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2';
 
 
-async function decodeEventBasicInfo(web3, event, timestampsCache) {
-  const timestamp = await timestampsCache.getBlockTimestamp(web3, event['blockNumber']);
+async function decodeEventBasicInfo(web3Wrapper, event, timestampsCache, addContract = true) {
```

Comment:
Add an extra argument to this function so that it can be reused by Matic codebase.